src/clients/Device.js

```diff
@@ -723,39 +541,68 @@ class Device extends EventEmitter {
     return Messages.getResponseType(request);
   };
 
+  // todo make return type annotation
+  getDescription = async (): Promise<*> => {
+    const isBusy = !this._isSocketAvailable(null);
+    if (isBusy) {
+      throw new Error('This core is locked during the flashing process.');
```

Comment:
put this in the most of newly added public functions...
i could wrap it into function like `_checkIsDeviceBusy()` - but it seems less clean